typescript/sdk/src/ism/EvmIsmModule.hardhat-test.ts

```diff
@@ -0,0 +1,720 @@
+  let exampleRoutingConfig: RoutingIsmConfig;
+  let mailboxAddress: Address;
+  let newMailboxAddress: Address;
+
+  let fundingAccount: SignerWithAddress;
+
+  const chain = TestChainName.test4;
+  let factoryAddresses: HyperlaneAddresses<ProxyFactoryFactories>;
+  let factoryContracts: HyperlaneContracts<ProxyFactoryFactories>;
+
+  beforeEach(async () => {
+    const [signer, funder] = await hre.ethers.getSigners();
+    fundingAccount = funder;
+    multiProvider = MultiProvider.createTestMultiProvider({ signer });
+
+    ismFactoryDeployer = new HyperlaneProxyFactoryDeployer(multiProvider);
+    const contractsMap = await ismFactoryDeployer.deploy(
+      multiProvider.mapKnownChains(() => ({})),
+    );
+
+    // get addresses of factories for the chain
+    factoryContracts = contractsMap[chain];
+    factoryAddresses = Object.keys(factoryContracts).reduce((acc, key) => {
+      acc[key] =
+        contractsMap[chain][key as keyof ProxyFactoryFactories].address;
+      return acc;
+    }, {} as Record<string, Address>) as HyperlaneAddresses<ProxyFactoryFactories>;
+
+    // legacy HyperlaneIsmFactory is required to do a core deploy
+    const legacyIsmFactory = new HyperlaneIsmFactory(
+      contractsMap,
+      multiProvider,
+    );
+
+    // mailbox
+    mailboxAddress = (
+      await new TestCoreDeployer(multiProvider, legacyIsmFactory).deployApp()
+    ).getContracts(chain).mailbox.address;
+
+    // new mailbox
+    newMailboxAddress = (
+      await new TestCoreDeployer(multiProvider, legacyIsmFactory).deployApp()
+    ).getContracts(chain).mailbox.address;
+
+    // example routing config
+    exampleRoutingConfig = {
+      type: IsmType.ROUTING,
+      owner: await multiProvider.getSignerAddress(chain),
+      domains: Object.fromEntries(
+        testChains
+          .filter((c) => c !== TestChainName.test4)
+          .map((c) => [c, randomMultisigIsmConfig(3, 5)]),
+      ),
+    };
+  });
+
+  // Wrapper for creating an ISM module
+  // include option for mailbox address override
+  async function createIsmModule({
+    config,
+    mailbox = mailboxAddress,
+  }: {
+    config: IsmConfig;
+    mailbox?: Address;
+  }): Promise<EvmIsmModule> {
+    return EvmIsmModule.create({
+      chain,
+      config,
+      deployer: ismFactoryDeployer,
+      factories: factoryAddresses,
+      mailbox,
+      multiProvider,
+    });
+  }
+
+  // Wrapper for checking whether ISM module matches a given config
+  // include option for mailbox address override
+  async function ismModuleMatchesConfig({
+    ism,
+    config,
+    mailbox = mailboxAddress,
+    configured = true,
+  }: {
+    ism: EvmIsmModule;
+    config: IsmConfig;
+    mailbox?: Address;
+    configured: boolean;
+  }): Promise<boolean> {
+    return moduleMatchesConfig(
+      chain,
+      ism.serialize().deployedIsm,
+      config,
+      multiProvider,
+      factoryContracts,
+      mailbox,
+      configured,
+    );
```

Comment:
I dont understand why we are using this in these tests

Reply:
usage of this fn and adding the `configured` arg are all downstream of this related discussion https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3759/files#r1609006616

assuming create() does the deploy()+update() then we can replace `moduleMatchesConfig` with deep equals comparator